### ISSUE TITLE
Alias request: "Microsoft.Compute/virtualMachines/storageProfile.dataDisks[*].encryptionSettings"

### ISSUE THREAD

**Issue by kutluomeroglu** [Azure/azure-policy#403]
### Scenario

- Azure Disk encryption extension is enabled on all azure VMswith volume type set to “all”.
- If a new disk is added to a VMs, azure disk encryption needs to be re-enabled for ARM to show data disk as encrypted.
- This alias would allow us to identify which disks are added after the disk encryption extension, so that we can go ahead and re-enable the ADE extension

### Alias

Include details about the alias being requested. Details on aliases are available [**here**](https://docs.microsoft.com/en-us/azure/governance/policy/concepts/definition-structure#aliases).
- What resource provider and resource type is involved?
Microsoft.Compute/virtualMachines
- What property on that resource type do you wish to expose?
EncryptionSettingsCollection
EncryptionSettingsCollection.Enabled

- A similar alias exists for OS disks:
Microsoft.Compute/virtualMachines/storageProfile.osDisk.encryptionSettings

### Example policy definition

Please include a sample policy definition that shows how you would like to use this new alias.

{
    "if": {
        "allOf": [
            {
                "field": "type",
                "equals": "Microsoft.Compute/virtualMachines"
            },
            {
                "anyOff": [
                    {
                        "field": "Microsoft.Compute/virtualMachines/storageProfile.dataDisks[*].EncryptionSettings",
                        "exists": false
                    },
                    {
                        "field": "Microsoft.Compute/virtualMachines/storageProfile.dataDisks[*].EncryptionSettings.enabled",
                        "equals": false
                    }
                ]
            }
        ]
    },
    "then": {
        "effect": "audit"
    }
}

### Note
For my scenario, I have achieved this using the definition below, however this alias still might be useful.
"policyRule": {
                    "if": {
                        "allOf": [
                            {
                                "field": "Microsoft.Compute/disks/diskState",
                                "equals": "Attached"
                            },
                            {
                                "field": "type",
                                "equals": "Microsoft.Compute/disks"
                            },
                            {
                                "field": "Microsoft.Compute/disks/encryptionSettingsCollection.enabled",
                                "notEquals": true
                            }
                        ]
                    },
                    "then": {
                        "effect": "audit"
                    }
                }

**Comment by mentat9** [comment]
The aliases you need are already published, but they are on another resource type. Note, the Get-AzPolicAlias cmdlet can be helpful in this situation. You don't even need to know the namespace, for example:

```
[C:\PS]-> Get-AzPolicyAlias -AliasMatch encryptionSettingsCollection | select -ExpandProperty Aliases | ?{ $_.Name -like '*encryptionSettingsCollection*' } | ft Name

Name
----
Microsoft.Compute/disks/encryptionSettingsCollection.enabled
Microsoft.Compute/disks/encryptionSettingsCollection.encryptionSettings[*].diskEncryptionKey.sourceVault.id
Microsoft.Compute/disks/encryptionSettingsCollection.encryptionSettings[*].diskEncryptionKey.sourceVault
Microsoft.Compute/disks/encryptionSettingsCollection.encryptionSettings[*].diskEncryptionKey.secretUrl
Microsoft.Compute/disks/encryptionSettingsCollection.encryptionSettings[*].diskEncryptionKey
Microsoft.Compute/disks/encryptionSettingsCollection.encryptionSettings[*].keyEncryptionKey.sourceVault.id
Microsoft.Compute/disks/encryptionSettingsCollection.encryptionSettings[*].keyEncryptionKey.sourceVault
Microsoft.Compute/disks/encryptionSettingsCollection.encryptionSettings[*].keyEncryptionKey.keyUrl
Microsoft.Compute/disks/encryptionSettingsCollection.encryptionSettings[*].keyEncryptionKey
Microsoft.Compute/disks/encryptionSettingsCollection.encryptionSettings[*]
Microsoft.Compute/disks/encryptionSettingsCollection.encryptionSettings
Microsoft.Compute/disks/encryptionSettingsCollection
Microsoft.Compute/snapshots/encryptionSettingsCollection.enabled
Microsoft.Compute/snapshots/encryptionSettingsCollection.encryptionSettings[*].diskEncryptionKey.sourceVault.id
Microsoft.Compute/snapshots/encryptionSettingsCollection.encryptionSettings[*].diskEncryptionKey.sourceVault
Microsoft.Compute/snapshots/encryptionSettingsCollection.encryptionSettings[*].diskEncryptionKey.secretUrl
Microsoft.Compute/snapshots/encryptionSettingsCollection.encryptionSettings[*].diskEncryptionKey
Microsoft.Compute/snapshots/encryptionSettingsCollection.encryptionSettings[*].keyEncryptionKey.sourceVault.id
Microsoft.Compute/snapshots/encryptionSettingsCollection.encryptionSettings[*].keyEncryptionKey.sourceVault
Microsoft.Compute/snapshots/encryptionSettingsCollection.encryptionSettings[*].keyEncryptionKey.keyUrl
Microsoft.Compute/snapshots/encryptionSettingsCollection.encryptionSettings[*].keyEncryptionKey
Microsoft.Compute/snapshots/encryptionSettingsCollection.encryptionSettings[*]
Microsoft.Compute/snapshots/encryptionSettingsCollection.encryptionSettings
Microsoft.Compute/snapshots/encryptionSettingsCollection


[C:\PS]->
```

**Comment by kutluomeroglu** [comment]
While I searched through Microsoft.compute/disks resource type to find the encrpytionSettingsCollection.enabled alias, I would probably search "encryption", "encryptionsettings" if I were to do a full search. While testing, I have come across these aliases:
- Microsoft.Compute/disks/encryptionSettings                                                                                                                                             
- Microsoft.Compute/disks/encryptionSettings.enabled   
which did not work as I expected them to work, maybe they should be removed as I am not sure if they are producing any meaningful results and causing confusion.
Consulting to Azure Resource Graph Explorer, I was able to figure out the properties that I was looking for were actually encryptionSettingsCollection. (encryptionSettingsCollection.enabled worked as I expected).
